pkg/metrics/service.go

```diff
@@ -22,42 +22,37 @@ import (
 
 	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
 	routev1 "github.com/openshift/api/route/v1"
-	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	v1 "k8s.io/api/core/v1"
 	k8serr "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	serviceLabelKey = "name"
 )
 
 var (
 	log = logf.Log.WithName("userMetrics")
 )
 
 // GenerateService returns the static service at specified port
-func GenerateService(port int32, portName string, serviceName string) (*v1.Service, error) {
+func GenerateService(port int32, portName string, serviceName, serviceNamespace string) (*v1.Service, error) {
 
 	// check if portname starts with "/"
 	if strings.HasPrefix(portName, "/") {
 		portName = portName[1:]
 	}
-	operatorName, err := k8sutil.GetOperatorName()
-	if err != nil {
-		return nil, err
-	}
-	namespace, err := k8sutil.GetOperatorNamespace()
-	if err != nil {
-		return nil, err
-	}
 
 	service := &v1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      serviceName,
-			Namespace: namespace,
-			Labels:    map[string]string{"name": operatorName},
+			Namespace: serviceNamespace,
+			Labels:    map[string]string{serviceLabelKey: serviceName},
```

Comment:
As written, the `serviceName` can now be empty. Do we want to omit the label in that case? Or make the name mandatory by making it a param to `NewBuilder` along with the `namespace` (and get rid of `WithServiceName`)?

Reply:
If an empty string is passed in deliberately then the configuration step would fail, right when the operator is starting up. So I don't think this should be a problem. But I can add a check for empty string.

Reply:
For consistency, IMO we should handle name and namespace the same way. Either both in the builder init, or both via `.With*()` chain funcs. Checking for empty strings either way would be goodness.